client/todos/reducer.ts

```diff
@@ -33,29 +31,29 @@ export default handleActions<IState, Todo>({
   },
 
   [EDIT_TODO]: (state: IState, action: Action<Todo>): IState => {
-    return <IState>state.map(todo =>
+    return state.map(todo =>
       todo.id === action.payload.id
-        ? assign(<Todo>{}, todo, { text: action.payload.text })
+        ? { ...todo, text: action.payload.text }
         : todo
     );
   },
 
   [COMPLETE_TODO]: (state: IState, action: Action<Todo>): IState => {
-    return <IState>state.map(todo =>
-      todo.id === action.payload.id ?
-        assign({}, todo, { completed: !todo.completed }) :
-        todo
+    return state.map(todo =>
+      todo.id === action.payload.id 
+        ? {...todo, completed: !todo.completed } 
+        : todo
     );
   },
 
   [COMPLETE_ALL]: (state: IState, action: Action<Todo>): IState => {
-    const areAllMarked = state.every(todo => todo.completed);
-    return <IState>state.map(todo => assign({}, todo, {
-      completed: !areAllMarked
-    }));
-  },
+     const areAllMarked = state.every(todo => todo.completed);
+     let x  = state.map(todo => ({...todo, completed: !areAllMarked }));
```

Comment:
maybe just `return state.map(todo => ({...todo, completed: !areAllMarked }))` ?

Reply:
Good point, I'll make the change locally and push. Thanks @snoosmoomrik